packages/next/src/build/define-env.ts

```diff
@@ -291,6 +289,18 @@ export function getDefineEnv({
     defineEnv[key] = userDefines[key]
   }
 
+  if (isNodeServer || isEdgeServer) {
+    const userServerDefines = config.compiler?.defineServer ?? {}
+    for (const key in userServerDefines) {
+      if (defineEnv.hasOwnProperty(key)) {
+        throw new Error(
+          `The \`compiler.serverDefine\` option is configured to replace the \`${key}\` variable. This variable is either part of a Next.js built-in or is already configured via the \`env\` option.`
```

Comment:
It's called defineServer though? There are a few places where you use "server define"